core/service/src/lib.rs

```diff
@@ -549,6 +569,10 @@ impl<Components> Future for Service<Components> where Components: components::Co
 	type Error = ();
 
 	fn poll(&mut self) -> Poll<Self::Item, Self::Error> {
+		if self.should_exit.load(Ordering::Relaxed) {
+			return Ok(Async::Ready(()));
```

Comment:
We should exit with a failure code here, so that service managers such as `systemd` or `daemontools` know that Substrate has failed and can respond accordingly.